### PR TITLE
[GLUTEN-7514][VL] Reorganize Dockerfiles and document how to build gluten in docker

### LINKED ISSUE #7514: [VL] Reorganize Dockerfiles and document how to build gluten in docker
### Description

See title.

### PR DESCRIPTION
**Pull request by PHILO-HE**
## What changes were proposed in this pull request?

* Place some Dockerfiles under dev/docker.
* Remove unnecessary docs like `docker_centos7.md` with some outdated content. User can just refer to `velox_weekly.yml` to install required tools for centos-7/8/9 and ubuntu-20.04/22.04.
* Add a document about how to build gluten in docker.

Related discussion: https://github.com/apache/incubator-gluten/pull/7511#issuecomment-2409722535

Fix #7514

### CODE REVIEW

File: .github/workflows/docker_image.yml [hunk 1]
```diff
@@ -21,6 +21,8 @@ on:
       - main
     paths:
       - '.github/workflows/docker_image.yml'
+      - 'dev/docker/Dockerfile.static-build'
```
[the remark this answers is not in the document]
Comment (PHILO-HE) [reply .github/workflows/docker_image.yml]: @zhouyuan, just changed to `Dockerfile.centos7-static-build`, `Dockerfile.centos8-dynamic-build`

**Comment by PHILO-HE** [comment]
@zhztheplayer, do you have any comment?

File: dev/docker/Makefile [hunk 2]
```diff
@@ -23,20 +23,27 @@ build: $(CCACHE_DIR) $(VCPKG_BINARY_CACHE_DIR) $(MAVEN_M2_DIR)
...
-		--file docker/Dockerfile.gha \
+		--file Dockerfile.centos7-static-build \
 		--tag "$(DOCKER_IMAGE)" \
 		--build-arg HTTPS_PROXY="" \
 		--build-arg HTTP_PROXY="" \
 		.
 
+docker-image-dynamic:
+	docker build \
+		--file Dockerfile.centos8-dynamic-build \
```
Comment (zhztheplayer) [inline dev/docker/Makefile:42]: Docker file is `Dockerfile.centos8-dynamic-build` while variable `DOCKER_IMAGE` is `apache/gluten:vcpkg-centos-7`. Something wrong here?

**Comment by zhztheplayer** [comment]
> @zhztheplayer, do you have any comment?

Direction of the change will look helpful to me. And I suggest to have the makefile tested locally before landing, as you can see all the images are using the same tag and they might override each other when building then all. I doubt we didn't test the file well before.

**Comment by PHILO-HE** [comment]
@zhztheplayer, just fixed Makefile and did some local tests. Could you take a look again?

**Review by zhouyuan — APPROVED** [review]

**Pull request merged**